validators.py

```diff
@@ -103,7 +103,8 @@ def validate(value, **_kwargs):
 
 def region_matches_treatment_code():
     def validate(region, **kwargs):
-        if region[0] != kwargs['row']['TREATMENT_CODE'][-1]:
+        if region.strip() and kwargs['row']['TREATMENT_CODE'].strip() and \
```

Comment:
maybe deserves a comment to explain the defensive coding, or you reckon it's self-explanatory enough? Looks fine to me, I have to say

Reply:
I think it's clear enough what the intention of the code is.